addon/PreviewProvider.js

```diff
@@ -359,15 +359,34 @@ PreviewProvider.prototype = {
   },
 
   /**
-   * Do some pre-processing on the links before inserting them into the metadata
+   * Do some pre-processing on the link before inserting it into the metadata
    * DB and adding them to the metadata cache
    */
-  processAndInsertMetadata(links) {
-    const processedLinks = this._processLinks(links);
-    this.insertMetadata(processedLinks);
+  processAndInsertMetadata(link, metadataSource) {
```

Comment:
I agree with @Mardak.  Because we are always taking in exactly one, take it in as a single entity and then wrap it in an array on line 366.